goopicsi.go

```diff
@@ -144,6 +161,37 @@ func NVMeControllerDisconnect(request *pb.NVMfRemoteControllerDisconnectRequest)
 	return &pb.NVMfRemoteControllerDisconnectResponse{}, nil
 }
 
+// CreateNVMeNamespace Creates a new NVMe namespace
+func CreateNVMeNamespace(id string, subSystemID string, volumeID string, hostID int32) (*pb.NVMeNamespace, error) {
+	if conn == nil {
+		err := dialConnection()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	client := pb.NewFrontendNvmeServiceClient(conn)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	resp, err := client.NVMeNamespaceCreate(ctx, &pb.NVMeNamespaceCreateRequest{
+		Namespace: &pb.NVMeNamespace{
+			Spec: &pb.NVMeNamespaceSpec{
+				Id:          &pbc.ObjectKey{Value: id},
+				SubsystemId: &pbc.ObjectKey{Value: subSystemID},
+				VolumeId:    &pbc.ObjectKey{Value: volumeID},
+				HostNsid:    hostID,
+			},
+		},
+	})
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	log.Println(resp)
+	return resp, nil
```

Comment:
you should not return `pb.NVMeNamespace` , maby string or int or error only...
otherwise this exposes internal implementation of the library and grpc...

Reply:
Fixed